src/include/btree_inline.h

```diff
@@ -916,6 +932,21 @@ __wt_page_parent_modify_set(WT_SESSION_IMPL *session, WT_REF *ref, bool page_onl
     return (0);
 }
 
+/*
+ * __wt_page_modify_update_timestamp --
```

Comment:
I understand why this is in btree_inline.h, but I'd put it in serial_inline.h. I *think* the compiler will be able to inline this check due to the include ordering, but we want to be sure given that this is called from the critical section of on-page assert.